### PR TITLE
Reorganised classes and modules.

### PR DESCRIPTION
**Pull request by HorGra**

### CODE REVIEW

File: database/src/main/java/com/kh/lolstats/DatabaseAccessor.java [hunk 1]
```diff
@@ -8,17 +8,24 @@
 
 public class DatabaseAccessor {
     private static DatabaseAccessor dbAccessor = null;
```
Comment (elbeyati) [inline database/src/main/java/com/kh/lolstats/DatabaseAccessor.java]: get rid of static. use object orientation

File: database/src/main/java/com/kh/lolstats/DatabaseAccessor.java [hunk 2]
```diff
@@ -26,26 +33,21 @@ public static DatabaseAccessor getDatabaseConnection(String url, String user, St
...
-    public Set<Long> getUncheckedSummonerIds() throws SQLException {
-        Set<Long> summonerIds = new HashSet<>();
-        try(Statement statement = connection.createStatement()){
-            ResultSet resultSet = statement.executeQuery("SELECT * FROM SUMMONERS;");
-            while(resultSet.next()){
-                summonerIds.add(resultSet.getLong("SUMMONER_ID"));
-            }
-            return summonerIds;
-        }
+    public void writeUpdates() throws SQLException {
```
Comment (elbeyati) [inline database/src/main/java/com/kh/lolstats/DatabaseAccessor.java]: no point having a method that calls another method

File: parser/src/main/java/com/kh/lolstats/util/Match.java [hunk 3]
```diff
@@ -128,4 +112,22 @@ public boolean getWinner() {
     public int[][] getRedChampsItems() {
         return redChampsItems;
     }
+
+    public String getMatchId() {
+        return matchId;
+    }
+
+    //    @Override
```
Comment (elbeyati) [inline parser/src/main/java/com/kh/lolstats/util/Match.java]: delete it

Comment (HorGra) [inline parser/src/main/java/com/kh/lolstats/util/Match.java]: deleted the commented overriden toString.

File: database/src/main/java/com/kh/lolstats/DatabaseAccessor.java (hunk 2 above)
Comment (HorGra) [inline database/src/main/java/com/kh/lolstats/DatabaseAccessor.java]: I didn't want to expose the Connection field.

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    @Override
+    public void run() {
+        try {
+            StopWatch stopWatch = new StopWatch();
+            stopWatch.start();
+            while (true) {
+                Thread.sleep(3000);
+                if(matchQueue.size() > 5){
+                    Set<Match> matchesToStore = new HashSet<>();
+                    matchesToStore.addAll(matchQueue);
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: use drainTo

File: persister/src/main/java/com/kh/lolstats/Persister.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+    private Connection initialiseConnection(String url, String user, String pass) throws SQLException, ClassNotFoundException {
+        Class.forName("com.mysql.jdbc.Driver");
+        Connection connection = DriverManager.getConnection(url, user, pass);
+        connection.setAutoCommit(false);
+        return connection;
+    }
+
+    public void onMatch(JsonNode matchDetails) throws InterruptedException, SQLException, IOException {
+        Match match = Match.parseFromMatchDetails(matchDetails);
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: you're still parsing on the parser thread.. you can put the JsonNode in the queue instead and do that work in the persister thread.

File: persister/src/main/java/com/kh/lolstats/ApiKeyAccessor.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package com.kh.lolstats;
+
+import java.util.Set;
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.TimeUnit;
+
+public class ApiKeyAccessor {
+    private static ApiKeyAccessor apiKeyAccessor = null;
+    private DelayQueue<DelayedApiKey> apiKeyQueue = new DelayQueue<>();
```
Comment (elbeyati) [inline persister/src/main/java/com/kh/lolstats/ApiKeyAccessor.java]: For future reference, always place all the static fields first, and then the other fields

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 4 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: changed a couple of fields and implemented drainTo

File: persister/src/main/java/com/kh/lolstats/Persister.java (hunk 5 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/Persister.java]: common decision to leave this in the parser for now.

File: persister/src/main/java/com/kh/lolstats/ApiKeyAccessor.java (hunk 6 above)
Comment (HorGra) [inline persister/src/main/java/com/kh/lolstats/ApiKeyAccessor.java]: done

**Pull request merged**